framework/tests/flow2/Purity.java

```diff
@@ -26,6 +28,19 @@ public PureClass() {}
         public SEClass() {}
     }
 
+    // sample test checking for deterministic
+    @MultipleRunDeterministic
```

Comment:
There are very few tests. Did you maybe not add additional test files? In particular tests that highlight the use of MRD for loops are important.